web/pdf_viewer.js

```diff
@@ -565,7 +565,7 @@ var PDFViewer = (function pdfViewer() {
         pageView.viewport.convertToViewportPoint(x + width, y + height)
       ];
       var left = Math.min(boundingRect[0][0], boundingRect[1][0]);
-      var top = Math.min(boundingRect[0][1], boundingRect[1][1]);
+      var top = Math.max(Math.min(boundingRect[0][1], boundingRect[1][1]), 0);
```

Comment:
I agree. The patch fixes the issue, but from the code it's not immediately clear why. Let's add a comment above this line that explains why we need this and mention `PR 6903` in the comment for reference.